access_error.go

```diff
@@ -34,10 +34,19 @@ func (f *Fosite) WriteAccessError(rw http.ResponseWriter, _ AccessRequester, err
 func (f *Fosite) writeJsonError(rw http.ResponseWriter, err error) {
 	rw.Header().Set("Content-Type", "application/json;charset=UTF-8")
 
-	rfcerr := ErrorToRFC6749Error(err)
+	// We dereference to make a copy and do not change passed err when it is already RFC6749Error.
+	rfcerr := *ErrorToRFC6749Error(err)
```

Comment:
I think this should be done everywhere around the codebase. I see often the pattern like:

```
rfcerr := ErrorToRFC6749Error(err)
rfcerr.Debug = ""
```

This in fact modifies the value in the original `err`. So if you try to log it to Prometheus or something similar afterwards, you do not have `Debug` anymore.

Reply:
See also this comment. Should I update this as well around?

Reply:
Yes, it makes sense for `ErrorToRFC6749Error` to return by value, not by ref. Please ensure that all method receivers operate on the value as well, except for the ones who modify the values.

Reply:
Maybe also rename this to `MakeRFC6749ErrorCopy`

Reply:
I made `Sanitize` method which does the copy and sanitization.